src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -249,7 +249,14 @@ public Collection<Object> createComponents(ExtensionsRunner runner) {
 
         Throttler throttler = new Throttler(getClock());
         ClientUtil clientUtil = new ClientUtil(environmentSettings, restClient(), throttler);
-        IndexUtils indexUtils = new IndexUtils(restClient(), clientUtil, sdkClusterService, indexNameExpressionResolver, javaAsyncClient());
+        IndexUtils indexUtils = new IndexUtils(
+            restClient(),
+            clientUtil,
+            sdkClusterService,
+            indexNameExpressionResolver,
+            javaAsyncClient(),
+            environmentSettings
```

Comment:
Similar comment here; we do have the settings at the end but can we just add an overloaded constructor that assumes empty settings if not passed so we don't have to rewrite existing constructor calls?